piet-scene/src/brush/color.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 The piet-gpu authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Also licensed under MIT license, at your choice.
+
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Default, Debug)]
+pub struct Color {
+    pub r: u8,
+    pub g: u8,
+    pub b: u8,
+    pub a: u8,
+}
+
+impl Color {
+    pub const fn rgb8(r: u8, g: u8, b: u8) -> Self {
+        Self { r, g, b, a: 255 }
+    }
+
+    pub const fn rgba8(r: u8, g: u8, b: u8, a: u8) -> Self {
+        Self { r, g, b, a }
+    }
+
+    pub fn rgb<C: Into<f64>>(r: C, g: C, b: C) -> Self {
+        Self::rgb8(
+            (r.into() / 255.0) as u8,
```

Comment:
It might be worth looking at the piet sources for this. This conversion neither rounds nor clamps to (0..=255) so I don't think it's quite ideal. Given that it's low level, it might not be important surfacing the float versions here.

Reply:
Thanks for not pointing out that all of these divisions should be multiplications anyway :) Clearly I never made use of the float versions. I'll go ahead and drop them.